### DISCUSSION TITLE
Steps Component

### DISCUSSION THREAD

**Discussion by yanthomasdev** [withastro/docs#885] [Ideas]
There are some places in the docs where we have step-by-step guides, the idea is to have a custom component that places each step and the following code/demo side by side. Here is the TailwindCSS docs component that has inspired this idea:

![image](https://user-images.githubusercontent.com/61414485/176314197-844582a3-db10-44c4-a1f0-025455c73a7e.png)

I first suggested this idea in [this Discord message](https://discord.com/channels/830184174198718474/853350631389265940/991461670423629985), Chris suggested a possible way of implementing this using Named Slots in Markdown that is worth taking a look.

- [x] I'm willing to PR this feature.

**Comment by ArmandPhilippot** [comment]
We're closing stale and outdated discussion to keep only the discussions still relevant to the current project version and/or the current docs state.

I'm closing this because a lot has happened since 2022! And I guess if you still want side-by-side step, this needs to be address in Starlight now! Thanks for the idea, Yan! 😄
